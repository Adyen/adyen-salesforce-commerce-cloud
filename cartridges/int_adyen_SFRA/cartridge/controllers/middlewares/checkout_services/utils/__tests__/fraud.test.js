"use strict";

var handleFraudDetection = require('../fraud');

var order;
var req;
var res;
var emit;
beforeEach(function () {
  order = {};
  req = {
    session: {
      privacyCache: {
        set: jest.fn()
      }
    }
  };
  res = {
    json: jest.fn()
  };
  emit = jest.fn();
});
describe('Fraud', function () {
  it('should return json response with error when fraud status is "fail"', function () {
    var isSuccessful = handleFraudDetection({
      status: 'fail'
    }, order, {
      req: req,
      res: res
    }, emit);
    expect(isSuccessful).toBeFalsy();
    expect(res.json.mock.calls).toMatchSnapshot();
    expect(emit).toHaveBeenCalledWith('route:Complete');
  });
  it('should succeed when fraud status is not "fail"', function () {
    var isSuccessful = handleFraudDetection({
      status: 'success'
    }, order, {
      req: req,
      res: res
    }, emit);
    expect(isSuccessful).toBeTruthy();
    expect(emit).toHaveBeenCalledTimes(0);
  });
});