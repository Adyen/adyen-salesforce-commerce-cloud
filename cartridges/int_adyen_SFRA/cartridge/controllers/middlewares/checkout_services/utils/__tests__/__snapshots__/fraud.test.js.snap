// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fraud should return json response with error when fraud status is "fail" 1`] = `
Array [
  Array [
    Object {
      "cartError": true,
      "error": true,
      "errorMessage": "mocked_error.technical",
      "redirectUrl": "[\\"Error-ErrorCode\\",\\"err\\",null]",
    },
  ],
]
`;
